docs/src/main/asciidoc/extension-authors-guide.adoc

```diff
@@ -17,7 +17,53 @@ Quarkus extensions add a new developer focused behavior to the core offering, an
 This means that metadata is only processed once at build time, which both saves on startup time, and also on memory
 usage as the classes etc that are used for processing are not loaded (or even present) in the runtime JVM.
 
-== Three Phases of Bootstrap and Quarkus Philosophy
+== Extension philosophy
+
+This section is a work in progress and gathers the philosophy under which extensions should be designed and written.
+
+=== Why an extension framework
+
+Quarkus’ mission is to transform your full applications and the libraries it uses into an optimal output for GraalVM.
+To do this you need to analyze and understand the full "closed world" of the application.
+Without the full and complete context, the best that can be achieved is partial and limited generic support.
+By using the Quarkus extension approach, we can bring high density to Java applications.
```

Comment:
not sure high-density is clear in this sentence.

Maybe says something like reduces memory and footprint.